packages/spark/sodasql/dialects/spark_dialect.py

```diff
@@ -169,6 +169,17 @@ def default_connection_properties(self, params: dict):
             'database': params.get('database', 'your_database')
         }
 
+    def safe_connection_data(self):
+        return [
+            self.type,
+            self.host,
```

Comment:
we will push this into `datasource_id`

Reply:
I would leave it at `host` unless for spark people always use the same host even between teams?

Reply:
Same as with snowflake, I do not know whether host is unique enough.  Keeping `type`, `host`, `port` and `database` for now.